### ISSUE TITLE
distribute docs

### ISSUE THREAD

**Issue by dimpase** [dimpase/primecountpy#5]
Should the pypi dist include doc sources?

**Comment by dimpase** [comment]
@mkoeppe

**Comment by mkoeppe** [comment]
If you want to make documentation installable, I would recommend to put it in a separate distribution package **primecountpy-doc** and upload that to PyPI.

**Comment by dimpase** [comment]
I'm thinking of using readthedocs, but for this one needs a working (pypi?) wheel package, or conda package.

Building pypi wheels is a dark art, unless there is a builder on GitHub actions...

This leaves Conda, but it hasn't packaged prinecount yet.

**Comment by dimpase** [comment]
I'm now able to build pypi wheels on Linux and macOS, see: https://github.com/dimpase/primecountpy/actions/runs/1521892016

Testing them automatically appears to be not too easy, at least I haven't found a way so far.
Anyway, I tested one of them on Linux, and one on macOS, manually, seems to work.
Any idea how to test them easily?

Before uploading them to pypi, I suppose
I should bump up the version and re-build them with the new version (as some toml, yaml etc. files were changed/removed).

**Comment by mkoeppe** [comment]
For testing the ones on Linux, I suppose you could duplicate the `ci-sage.yml` workflow to test the wheels on various platforms...

**Comment by dimpase** [comment]
automatic re-building readthedocs (using a pypi wheel for primecountpy) is working now too, check out https://primecountpy.readthedocs.io/en/latest/

**Comment by mkoeppe** [comment]
On macOS I am getting:
```
$ pip install --user --only-binary :all:  primecountpy
Collecting primecountpy
  Downloading primecountpy-0.0.0-cp39-cp39-macosx_10_9_x86_64.whl (174 kB)
     |████████████████████████████████| 174 kB 2.4 MB/s 
Requirement already satisfied: Cython in /Users/mkoeppe/Library/Python/3.9/lib/python/site-packages (from primecountpy) (0.29.24)
Installing collected packages: primecountpy
Successfully installed primecountpy-0.0.0
WARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
You should consider upgrading via the '/usr/local/opt/python@3.9/bin/python3.9 -m pip install --upgrade pip' command.
egret:~/s/sage/sage-rebasing/worktree-gcc11 (t/32432/modularization_of_sagelib__break_out_a_separate_package_sagemath_polyhedra $%>)$ python3
Python 3.9.7 (default, Oct 13 2021, 06:45:31) 
[Clang 13.0.0 (clang-1300.0.29.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import primepy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'primepy'
>>> import primecountpy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mkoeppe/Library/Python/3.9/lib/python/site-packages/primecountpy/__init__.py", line 1, in <module>
    from .primecount import *
  File "primecountpy/primecount.pyx", line 1, in init primecountpy.primecount
  File "/Users/mkoeppe/Library/Python/3.9/lib/python/site-packages/cysignals/__init__.py", line 1, in <module>
    from .signals import AlarmInterrupt, SignalError, init_cysignals  # noqa
ImportError: dlopen(/Users/mkoeppe/Library/Python/3.9/lib/python/site-packages/cysignals/signals.cpython-39-darwin.so, 2): Library not loaded: /usr/local/opt/pari/lib/libpari-gmp.dylib
  Referenced from: /Users/mkoeppe/Library/Python/3.9/lib/python/site-packages/cysignals/signals.cpython-39-darwin.so
  Reason: image not found
```

**Comment by mkoeppe** [comment]
Also, I don't think you need `install_requires=["Cython"]`.

**Comment by mkoeppe** [comment]
But as the errors above show, you do need `cysignals` at runtime. So this needs to go in `install_requires`.

**Comment by mkoeppe** [comment]
After this change, you can remove `cysignals` from `docs/requirements.txt`.

**Comment by dimpase** [comment]
OK, thanks, I suppose I tested with `libpari` installed and available.

**Comment by mkoeppe** [comment]
https://github.com/sagemath/cysignals/pull/149 will remove the cypari dependency

**Comment by dimpase** [comment]
> On macOS I am getting:

fixed in 0.1.0.

(I even tested wheels in a clean venv)
